Expensify/App
# [No QA] Add deployer GH handle to deploy messages

@AndrewGable Will you please review this?

### Details
This adds the deployer's GH handle to deploy messages, where the deployer is defined as:

1. For production deploys, the person who closed the checklist to trigger the deploy.
1. For normal staging deploys, the person who merged the PR.
1. For "automatic" CPs (using the `CP Staging` label), the person who merged the PR.
1. For manual CPs using the GH UI, the person who triggered the workflow. (We are going to save this by including the actor's GH handle in the CP branch name.

### Fixed Issues
$ https://github.com/Expensify/Expensify/issues/168264

### Tests
1. Ensure that the deploy checklist is unlocked.
1. Merge this PR, and after the staging deploy completes, verify that it gets a "Deployed to staging by _\<whoever merged it>_" comment.
1. Lock the deploy checklist.
1. Create and merge another PR with the `CP Staging` label. Verify that it gets a "Cherry-picked to staging by _\<whoever merged it>_" comment.
1. Manually CP a PR using the GH UI. For testing purposes, the person who manually CPs the PR should be different than the person who merged it. After the staging deploy completes. verify that it gets a "Cherry-picked to staging by _\<whoever manually triggered the CP>_" comment.

### QA Steps
None.

### Tested On

n/a, GH only. But I did do a _lot_ of incremental testing using a local octokit instance while developing.
